bedrock/base/templates/404.html

```diff
@@ -21,8 +21,19 @@
 
 {% block content %}
   <main class="mzp-l-content message-section">
-    <h1 class="error-title">{{ ftl('not-found-page-sorry-we-cant-find-that-page') }}</h1>
-    <p>{{ftl('not-found-page-were-all-about-a-healthy-internet') }}</p>
+
+    {% if "/careers/" in request.path %}
+      <h1 class="error-title">Sorry, we can’t find that job posting</h1>
+      <p>
+        The link you followed doesn’t seem to be for an open position.
+        <a href="{{ url('careers.listings') }}">
+          Check out our current job listings.
+        </a>
+      </p>
+    {% else %}
+      <h1 class="error-title">{{ ftl('not-found-page-sorry-we-cant-find-that-page') }}</h1>
+      <p>{{ftl('not-found-page-were-all-about-a-healthy-internet') }}</p>
+    {% endif %}
```

Comment:
Yeah it does. Wagtail maps URL paths to specific pages in the CMS page tree via a special view called `wagtail.views.serve`, which is included in the main URLConf via the `wagtail_urls` module. This, by design, works best at the bottom of the URLConf, mainly (IIRC) to avoid stopping any hard-coded URL paths (either in a project's other apps or in third-party apps) from matching.

The trade-off with this approach is that if an earlier view raises a 404 (e.g. the custom careers 404) , `wagtail.views.serve` still gets called, fails to match and then raises a (regular) 404 instead, losing the custom message. As such, this was a way to ensure it still happens on careers post pages.

Does that thinking stand up for you?

(Looking at it now, we could/should make the `if` test a little more specific, so that we don't get that 404 on any page under /careers/ (eg a typoed URL), not just closed job listings that now 404.)